hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/cache/FederationGuavaCache.java

```diff
@@ -60,7 +63,7 @@ public void initCache(Configuration pConf, FederationStateStore pStateStore) {
 
     // Initialize Cache.
     cache = CacheBuilder.newBuilder().expireAfterWrite(cacheTimeToLive,
-        TimeUnit.MILLISECONDS).build();
+        TimeUnit.MILLISECONDS).maximumSize(cacheEntityNums).build();
```

Comment:
cacheTimeToLive = pConf.getInt(YarnConfiguration.FEDERATION_CACHE_TIME_TO_LIVE_SECS,
        YarnConfiguration.DEFAULT_FEDERATION_CACHE_TIME_TO_LIVE_SECS);

TimeUnit.MILLISECONDS -> TimeUnit.SECONDS

Reply:
Thanks for your suggestion! I will fix it.